core/trino-main/src/main/java/io/trino/operator/DriverYieldSignal.java

```diff
@@ -54,6 +57,7 @@ public synchronized void setWithDelay(long maxRunNanos, ScheduledExecutorService
     {
         checkState(yieldFuture == null, "there is an ongoing yield");
         checkState(!isSet(), "yield while driver was not running");
+        checkState(!terminationStarted, "yield after termination started");
```

Comment:
Maybe simply `if(terminationStarted) return` to avoid unnecessary (though low likelyhood) exceptions?

Reply:
With the current ordering of operations in `Driver#close()` this should not be possible, but given your suggestion of doing this unconditionally before interrupting then this could throw an exception so I'll make both changes suggested.